packages/core/src/users.ts

```diff
@@ -0,0 +1,15 @@
+import AbstractMessaging from "../../messaging-interface/src/index";
+
+export abstract class User extends AbstractMessaging {
+
+  abstract createUser(userInput: object | any): void;
+  abstract getUser(userInput: object | any): void;
+  abstract fetchChats(fetchChatInput: object | any): void;
```

Comment:
These abstract functions are in the messaging class and not needed here. So the messaging-core would extend this "User" and have these functions defined. 
Then the specific messaging package can extend on these functions.